tests/analyses/test_checks.py

```diff
@@ -0,0 +1,97 @@
+from datetime import datetime
+
+import pytest
+
+from virtool.analyses.checks import (
+    check_if_analysis_modified,
+    check_if_analysis_ready,
+    check_analysis_workflow,
+    check_if_analysis_running,
+    check_analysis_nuvs_sequence,
+)
+from virtool.data.errors import (
+    ResourceNotModifiedError,
+    ResourceConflictError,
+    ResourceNotFoundError,
+)
+
+
+@pytest.mark.parametrize("error", [True, False])
+async def test_check_if_analysis_modified(error, static_time):
+    analysis = {"id": "baz", "created_at": static_time}
+    if_modified_since = static_time
+
+    if error:
+        with pytest.raises(ResourceNotModifiedError):
+            await check_if_analysis_modified(if_modified_since, analysis)
+        return
+
+    if_modified_since = datetime(2016, 12, 25, 8, 0, 0)
+    assert await check_if_analysis_modified(if_modified_since, analysis) is None
+
+
+@pytest.mark.parametrize("error", [True, False])
+async def test_check_if_analysis_ready(error):
+
+    jobs_api_flag = True
+    ready = True
+
+    if error:
+        with pytest.raises(ResourceConflictError):
+            await check_if_analysis_ready(jobs_api_flag, ready)
+        return
+
+    jobs_api_flag = False
+    assert await check_if_analysis_ready(jobs_api_flag, ready) is None
```

Comment:
You're going to refactor this, but why not just:
```suggestion
    if error:
        with pytest.raises(ResourceConflictError):
            await check_if_analysis_ready(True, True)
        return

    jobs_api_flag = False
    assert await check_if_analysis_ready(False, True) is None
```
You do this quite a bit in this PR.